scripts/split_repo_migration/git-migrate-history.sh

```diff
@@ -0,0 +1,167 @@
+echo "Created working directory: ${WORKDIR}"
+
+pushd "${WORKDIR}"  # cd into workdir
+
+echo "Cloning source repository: ${SOURCE_REPO}"
+git clone "git@github.com:${SOURCE_REPO}.git" source-repo
+
+pushd source-repo
+git remote remove origin
+
+# prune only files within the specified directory
+if [[ ! -z "${SOURCE_PATH}" ]]
+then
+  echo "Pruning commits only including path: ${SOURCE_PATH}"
+  git filter-branch \  # rewrites history... unsafe
+    --prune-empty \
+    --subdirectory-filter "${SOURCE_PATH}"
+fi
+
+if [[ ! -z "${IGNORE_FOLDERS}" ]]
+then
+  echo "Ignoring folder: ${IGNORE_FOLDERS}"
+  mkdir -p "${WORKDIR}/filtered-source"
+  FOLDERS=$(echo ${IGNORE_FOLDERS} | tr "," " ")
+  # remove files/folders we don't want
+  FILTER="(rm -rf ${FOLDERS} || true)"
+  if [[ ! -z "${KEEP_FILES}" ]]
+  then
+    KEEP_FILES_SPACES=($(echo ${KEEP_FILES} | tr "," " "))
+    LAST_ELEMENT=$(( ${#KEEP_FILES_SPACES[@]} - 1 ))
+    KEEP_FILE_COMMANDS=""
+    for file in "${KEEP_FILES_SPACES[@]}"
+    do
+      if [[ $file == "${KEEP_FILES_SPACES[$LAST_ELEMENT]}" ]]
+      then
+        KEEP_FILE_COMMANDS+="git checkout -- ${file} 2>/dev/null || true"
+      else 
+        KEEP_FILE_COMMANDS+="git checkout -- ${file} 2>/dev/null || true; "
+      fi   
+    done
+    # restore files to keep, silence errors if the file doesn't exist
+    FILTER="${FILTER}; ${KEEP_FILE_COMMANDS}"
+  fi
+  git filter-branch \
+    --force \
+    --prune-empty \
+    --tree-filter "${FILTER}"
+fi
+
+TARGET_PATH="packages/$(jq -r '.distribution_name' .repo-metadata.json)"  # -r removes quotes around the name.
+
+# reorganize the filtered code into the desired target locations
+if [[ ! -z "${TARGET_PATH}" ]]
+then
+  echo "Moving files to destination path: ${TARGET_PATH}"
+  git filter-branch \
+    --force \
+    --prune-empty \
+    --tree-filter \
+      "shopt -s dotglob; mkdir -p ${WORKDIR}/migrated-source; mv * ${WORKDIR}/migrated-source; mkdir -p ${TARGET_PATH}; mv ${WORKDIR}/migrated-source/* ${TARGET_PATH}"
+fi
+
+# back to workdir
+popd
+
+# merge histories
+pushd $TARGET_REPO
+
+REMOTE="remote.${SOURCE_REPO}"
+git remote add --fetch ${REMOTE} ${WORKDIR}/source-repo
+git checkout -B "${BRANCH}"
+git merge --allow-unrelated-histories ${REMOTE}/main --no-edit
+
+echo "Success"
+
+popd # back to workdir
+
+# Do a diff between source code split repo and migrated code.
+git clone "git@github.com:${SOURCE_REPO}.git" source-repo-validation  # Not ideal to clone again.
+rm -rf source-repo-validation/.git  # That folder is not needed for validation.
+
+DIFF_FILE="${WORKDIR}/diff.txt"
+if diff -r "${TARGET_REPO}/${TARGET_PATH}" source-repo-validation > "${DIFF_FILE}" ; then
+  echo "No diff"
+else
+  echo "Diff non-empty. See ${DIFF_FILE}"
+  $EXIT 1
+fi
+
+pushd "${TARGET_REPO}"  # To target repo
+
+# git push -u origin "${BRANCH}" --force
+
+# # create pull request
+# if gh --help > /dev/null
+# then
+#   gh pr create --title "migrate code from ${SOURCE_REPO}"
```

Comment:
This command was prompting me for PR info, which we don't want if we run this in a batch script. Let's figure out how to pass everything on the command line.

Reply:
We could do this as a follow-up PR, IMO.

Reply:
I think just adding `--body "Migrate ${SRC_API_NAME} into ${TARGET_PATH}"` should do it.

Reply:
Also, we need a conventional commit title format....